### PR TITLE
fixing custom metric

### PR DESCRIPTION
**Pull request by liususan091219**
fixing the custom metric bug

### CODE REVIEW

File: test/nlp/test_autohf_custom_metric.py [hunk 1]
```diff
@@ -15,11 +15,25 @@ def toy_metric(
...
+
+    trainer = estimator._model
+
+    trainer_compute_metrics_cache = trainer.compute_metrics
+    trainer.compute_metrics = None
+
+    metrics = trainer.evaluate(eval_dataset)
+    trainer.compute_metrics = trainer_compute_metrics_cache
+
+    return 0, metrics
```
Comment (sonichi) [inline test/nlp/test_autohf_custom_metric.py:36]: are you sure you want to return 0 as the metric to minimize?

File: flaml/model.py [hunk 2]
```diff
@@ -367,12 +368,20 @@ def _init_hpo_args(self, automl_fit_kwargs: dict = None):
...
         from .nlp.utils import tokenize_text
 
-        if X.dtypes[0] == "string":
+        is_str = False
+        for each_type in ["string", "str"]:
+            try:
+                is_str = is_str or (X.dtypes[0] == each_type)
+            except TypeError:
+                print("Type {} not understood".format(each_type))
+                pass
```
[the remark this answers is not in the document]
Comment (liususan091219) [reply flaml/model.py:380]: It’ll throw the error

[the remark this answers is not in the document]
Comment (liususan091219) [reply flaml/model.py:380]: On windows pytest

File: flaml/model.py [hunk 3]
```diff
@@ -367,12 +368,20 @@ def _init_hpo_args(self, automl_fit_kwargs: dict = None):
             setattr(custom_hpo_args, key, val)
         self.custom_hpo_args = custom_hpo_args
 
-    def _preprocess(self, X, y=None, task=None, **kwargs):
+    def _preprocess(self, X, y=None, **kwargs):
         from .nlp.utils import tokenize_text
 
-        if X.dtypes[0] == "string":
+        is_str = False
+        for each_type in ["string", "str"]:
```
[the remark this answers is not in the document]
Comment (liususan091219) [reply flaml/model.py:375]: yea one of them will throw an execption on windows so cannot use one line. 

If we lay out two try statements that's less concise than a for loop.

File: flaml/model.py (hunk 2 above)
Comment (sonichi) [inline flaml/model.py:380]:
```suggestion
        is_str = X.dtypes[0] in ("string", "str")
```
Won't this work?

**Review by sonichi — APPROVED** [review]

**Pull request merged**